### DISCUSSION TITLE
Multi devices join in parallel by OTAA mode

### DISCUSSION THREAD

**Discussion by QINLIMIN** [Lora-net/LoRaMac-node#1174] [Q&A]
Hi, we got some issues when we setup a few tens devices join to NS in parallel by OTAA mode:
1, When the devices power on one by one, all devices join success rate were 100%. But the success rate just around 70% when they power on at the same time. Means some devices cannot joined when they send join request at one time.
2, Due to our devices are power by power grid, it not worry about the battery consumption. So we want the devices rejoin forever when they join fail, We have tried to disable duty cycle, but it not work, it still have "duty cycle restrict" error which as LoRaWAN 1.0.4 specification. 
Please help give some suggestions, how to meet the above requirements. thank you.

**Comment by mluis1** [comment]
The LoRaWAN specification chapter "7  Retransmissions Backoff" is present in order to avoid the described situation.
This project implements everything that is required to support what is described in this chapter.

In order to improve the randomization this project uses the radio interface to generate a random number which will be used as seed for the pseudo-random generator. By doing so we try to ensure as much as possible that every end-device gets a different pseudo-random sequence.

The provided examples further improve the situation by randomizing the end-device transmission. Please refer to the following:
https://github.com/Lora-net/LoRaMac-node/blob/05353ff70bec5349a26b23a02168accb0ea03ad9/src/apps/LoRaMac/periodic-uplink-lpp/NucleoL073/main.c#L55-L64
and
https://github.com/Lora-net/LoRaMac-node/blob/05353ff70bec5349a26b23a02168accb0ea03ad9/src/apps/LoRaMac/periodic-uplink-lpp/NucleoL073/main.c#L273-L274

While making this answer I noticed that the example could be further improved by changing following function implementation from
https://github.com/Lora-net/LoRaMac-node/blob/05353ff70bec5349a26b23a02168accb0ea03ad9/src/apps/LoRaMac/periodic-uplink-lpp/NucleoL073/main.c#L530-L542

To:

```c
/*!
 * Function executed on TxTimer event
 */
static void OnTxTimerEvent( void* context )
{
    TimerStop( &TxTimer );

    IsTxFramePending = 1;

    // Schedule next transmission
    TxPeriodicity = APP_TX_DUTYCYCLE + randr( -APP_TX_DUTYCYCLE_RND, APP_TX_DUTYCYCLE_RND );
    TimerSetValue( &TxTimer, TxPeriodicity );
    TimerStart( &TxTimer );
}
```
